### PR TITLE
Migration to slog "in place"

### PR DESCRIPTION
**Pull request by kinnou02**
This PR doesn't introduce any of the advantage of slog but add the
possibility to use json as output format

Another PR will exploit slog

### CODE REVIEW

File: libs/mimir/src/lib.rs [hunk 1]
```diff
@@ -48,39 +54,50 @@ pub mod objects;
...
-                "[{time}] [{lvl}] [{loc}] {msg}\n",
-                time = Local::now(),
-                lvl = record.level(),
-                loc = record.module_path().unwrap_or("unknown"),
-                msg = record.args()
-            )
-        });
+pub fn logger_init() -> (slog_scope::GlobalLoggerGuard, ()) {
+    if let Ok(s) = env::var("RUST_LOG_JSON") {
+        let mut drain = slog_json::Json::new(std::io::stdout())
```
Comment (TeXitoi) [inline libs/mimir/src/lib.rs]: stderr as in the other branch?

File: libs/mimir/src/lib.rs [hunk 2]
```diff
@@ -48,39 +54,50 @@ pub mod objects;
...
+{
+    //by default we log for info
+    let builder = slog_envlogger::LogBuilder::new(drain).filter(None, slog::FilterLevel::Info);
+    let builder = if let Ok(s) = env::var("RUST_LOG") {
+        builder.parse(&s)
+    } else {
+        builder
+    };
+    let drain = slog_async::Async::new(builder.build())
+        .chan_size(100 * 1024)
```
Comment (TeXitoi) [inline libs/mimir/src/lib.rs]: That's a lot

File: libs/mimir/src/lib.rs [hunk 3]
```diff
@@ -48,39 +54,50 @@ pub mod objects;
...
+        .build();
+
+    let log = slog::Logger::root(drain.fuse(), slog_o!());
+    let _scope_guard = slog_scope::set_global_logger(log);
+    let _log_guard = slog_stdlog::init().unwrap();
```
Comment (TeXitoi) [inline libs/mimir/src/lib.rs]: please no _ at the begining of a variable name if it is used after.

File: libs/mimir/src/lib.rs (hunk 2 above)
Comment (kinnou02) [reply libs/mimir/src/lib.rs]: It is, but I keep having message dropped with de default (256) settings.
I run bragi more or less like we will in production, log in json format with everything in error except mimir and bragi log at info 
```
RUST_LOG_JSON=simple RUST_LOG="error,rs_es=error,iron=error,bragi=info,mimir=info" ./target/release/bragi
```
In do a simple `ab` on it, 1 thread and 1000 request
```
ab -c1 -n1000 'http://localhost:4000/autocomplete?q=1%20rue%20de%20l%20%C3%A9glise%20-%2069480%20moranc%C3%A9'
```
and I quickly get these log:
```
{"msg":"slog-async: logger dropped messages due to channel overflow","level":"ERRO","ts":"2018-02-26T08:43:01.169377935+01:00","module":"slog_async","count":44}
{"msg":"slog-async: logger dropped messages due to channel overflow","level":"ERRO","ts":"2018-02-26T08:43:01.169546034+01:00","module":"slog_async","count":159}
{"msg":"slog-async: logger dropped messages due to channel overflow","level":"ERRO","ts":"2018-02-26T08:43:01.169712304+01:00","module":"slog_async","count":82}
{"msg":"slog-async: logger dropped messages due to channel overflow","level":"ERRO","ts":"2018-02-26T08:43:01.169864256+01:00","module":"slog_async","count":70}

```

Same thing with a channel size of 1024.
But I agree than a bigger queue isn't the answer to the problem.

Comment (TeXitoi) [reply libs/mimir/src/lib.rs]: Even in release? Seems strange, hyper use the log crate, and the log crate disable trace in release.

Comment (kinnou02) [reply libs/mimir/src/lib.rs]:
```
RUST_LOG_JSON=simple RUST_LOG="trace" ./target/release/bragi 
listening on 127.0.0.1:4000
{"msg":"threads = 64","level":"DEBG","ts":"2018-02-26T10:35:56.457625271+01:00","module":"hyper::server"}
{"msg":"Incoming stream","level":"DEBG","ts":"2018-02-26T10:35:59.768685714+01:00","module":"hyper::server"}
{"msg":"get_buf []","level":"TRCE","ts":"2018-02-26T10:35:59.768734763+01:00","module":"hyper::buffer"}
{"msg":"read_into_buf buf[0..4096]","level":"TRCE","ts":"2018-02-26T10:35:59.768774370+01:00","module":"hyper::buffer"}
{"msg":"get_buf [u8; 4096][0..150]","level":"TRCE","ts":"2018-02-26T10:35:59.768822525+01:00","module":"hyper::buffer"}

```

Comment (TeXitoi) [reply libs/mimir/src/lib.rs]:
```diff
diff --git a/Cargo.toml b/Cargo.toml
index f0f6fe9..4be4083 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -7,6 +7,7 @@ build = "build.rs"
 [workspace]
 
 [dependencies]
+log = { version = "0.4", features = ["release_max_level_info"] }
 slog = { version = "2.1", features = ["max_level_trace", "release_max_level_info"]}
 slog-scope = "4.0"
 slog-envlogger = "2.1.0"
```

Comment (kinnou02) [reply libs/mimir/src/lib.rs]: I have disabled Trace log, after that I don't overload the channel even with  20 concurrent requests

**Pull request merged**